generators/go/internal/generator/sdk/core/stream.go

```diff
@@ -162,11 +162,15 @@ func (s *scannerStreamReader) parse(bytes []byte) (int, []byte, error) {
 		}
 	}
 	data := bytes[start:]
-	if i := strings.Index(string(data), s.options.delimiter); i >= 0 {
-		data = data[:i+len(s.options.delimiter)]
+	delimIndex := strings.Index(string(data), s.options.delimiter)
+	if delimIndex < 0 {
+		return len(bytes), data, nil
 	}
-	n := start + len(data) + len(s.options.delimiter)
-	return n, data, nil
+	endIdx := delimIndex + len(s.options.delimiter)
+	parsedData := data[:endIdx]
```

Comment:
Do we need to add any additional handling for the fix we made in TypeScript [here](https://github.com/fern-api/fern/pull/5471), or is this working well already?

Reply:
It is working well -- the logic there ends when no additional delimiter is found.